internal/pkg/ntp/ntp.go

```diff
@@ -394,13 +394,22 @@ func (syncer *Syncer) queryServer(server string) (*ntp.Response, error) {
 	return resp, err
 }
 
+// log2i returns 0 for v == 0 and v == 1.
+func log2i(v uint64) int {
+	if v == 0 {
+		return 0
+	}
+
+	return 63 - bits.LeadingZeros64(v)
```

Comment:
this is the real fix, previous code was `s/63/64`.

the code in systemd for comparison: https://github.com/systemd/systemd/blob/6639ac474eb7a5325a72a3d7370492792dd00bc0/src/fundamental/logarithm.h#L39-L40